### PR TITLE
Fix issues related to setting up development env

### PR DESCRIPTION
**Pull request by mo-dkrz**
When I was setting up a development environment based on he instruction in README, I observed a couple of minor issues which I propose them as a PR:

Makefile: generating certs via the `dev-utils` instead of `run_server`
utils.py: changing `appdirs` to `platformdirs` to be able to work on diff OSs on python3.12 [ref](https://github.com/conda/conda/issues/13306)
posix.py: get the exception of cfgrib out of coverage as rarely it's gonna happen

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -7,7 +7,7 @@ install: prepare
 prepare:
 	python3 -m pip install cryptography tox
 	mkdir -p dev-env/certs
-	python3 run_server.py --gen-certs
+	python3 dev-env/config/dev-utils.py gen-certs
 
 lint:
 	tox -e lint types
```
[the remark this answers is not in the document]
Comment (eelucio) [reply Makefile:13]: Also, `make docs` and `make release` do not work.

should  it be?

```toml
.PHONY: docs all release 
```
instead of 
```toml
.PHONY: all release 
```

and 

```toml
docs:
       tox -e docs

release:
       tox -e release
```

instead of

```toml
docs: tox -e docs                                                                                                                           
                                                                                                                                            
release: tox -e release  
```

**Comment by eelucio** [comment]
thanks @mo-dkrz 

in the main branch I was getting the following when running `make install`:
```
mkdir -p dev-env/certs
python3 run_server.py --gen-certs
Trying http://localhost:8080/realms/freva/.well-known/openid-configuration...<urlopen error [Errno 111] Connection refused>
...
Trying http://localhost:8080/realms/freva/.well-known/openid-configuration...<urlopen error [Errno 111] Connection refused>

Open ID connect service is not up.
Traceback (most recent call last):
  File "/home/etor/work/freva/github/freva-nextgen/run_server.py", line 142, in <module>
    cli()
  File "/home/etor/work/freva/github/freva-nextgen/run_server.py", line 129, in cli
    run(
  File "/home/etor/miniconda3/envs/freva-nextgen/lib/python3.12/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/home/etor/miniconda3/envs/freva-nextgen/bin/python3', 'dev-env/config/dev-utils.py', 'oidc', 'http://localhost:8080/realms/freva/.well-known/openid-configuration']' returned non-zero exit status 1.
make: *** [Makefile:10: prepare] Error 1
```
but now I can install it fully.

One thing -> could you add 

```
    conda create -n freva-nextgen -c conda-forge cfgrib
    conda run -n freva-nextgen make install
```

instead of

```
    conda create -n freva-nextgen cfgrib
    conda run -n freva-nextgen make install
```

in the README? it could happen that not everyone (like me) changed their conda default channel to conda-forge.

---

I am following the README file with the tests, for some reason it took me some time to stablish the connection with the keycloak, 

```
python run_server.py -c api_config.toml --debug --dev -p 7777 -f
Trying http://localhost:8080/realms/freva/.well-known/openid-configuration...Remote end closed connection without response
...
```
 but eventually it stablished the connection and I could run

`tox -p 3`

and also e.g. `make docs`  without issues.

**Comment by mo-dkrz** [comment]
@eelucio thanks,

1. about the first comment, I thought a bit, and as it is a development section, I think adding ` -c conda-forge` to the conda creating env command won't be necessary.

File: Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (mo-dkrz) [reply Makefile:13]: And I had a look, in dev section of README we have no `make lint` and other commands and this PR is about making the dev part works, so if you are okey, let's submit another PR for this. what do u think?

**Comment by mo-dkrz** [comment]
@antarcticrainforest when you have time I will be appreciated to have a look a this

File: Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (antarcticrainforest) [reply Makefile:13]: It should be `tox -e lint,types`

[the remark this answers is not in the document]
Comment (mo-dkrz) [reply Makefile:13]: do i have to change it here or in another PR?

File: freva-data-portal-worker/src/data_portal_worker/utils.py [hunk 2]
```diff
@@ -7,7 +7,7 @@
 from socket import gethostname
 from typing import Optional
 
-import appdirs
```
Comment (antarcticrainforest) [inline freva-data-portal-worker/src/data_portal_worker/utils.py:10]: Just out of curiousity why was this necessary?

Comment (mo-dkrz) [reply freva-data-portal-worker/src/data_portal_worker/utils.py:10]: When I was setting up development env on my machine i faced an issue regarding using this lib and when I searched a bit, I found this lib is gonna be deprecated in the future and be replaced with `platformdirs`; as an example conda already has been started replacing ...

https://github.com/conda/conda/issues/13306

File: Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (antarcticrainforest) [reply Makefile:13]: No, preference!

File: freva-data-portal-worker/src/data_portal_worker/utils.py (hunk 2 above)
Comment (antarcticrainforest) [reply freva-data-portal-worker/src/data_portal_worker/utils.py:10]: Ok, I see!

**Review by antarcticrainforest — APPROVED** [review]
LGTM!
